### PR TITLE
Add warning about request forgery and client-side migration. Fixes #4086

### LINKED ISSUE #4086: Add warning about request forgery and client-side migration
I find Martin's argument that QUIC servers are able to control their topology well enough to prevent request forgery attacks somewhat persuasive. However:

1. I think the document should be clearer that if you can't do this you can't deploy QUIC as a server.
2. We should add text indicating that a candidate extension to allow migration by the client would need to deal with this (likely by adding one of the countermeasures we decided not to ass). This is a very long specification and otherwise the institutional knowledge about this issue is likely to be lost.

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by ekr** [comment]
I also think it might be worthwhile to move much of this section to security considerations, leaving only the requirements here. @martinthomson WDYT?

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -6440,8 +6440,19 @@ This section also describes limited countermeasures that can be implemented by
...
+{{request-forgery-with-spoofed-migration}} is quite powerful and does
+not have adequate countermeasures, QUIC server implementations should
+assume that attackers can cause them to generate arbitrary UDP
+payloads to arbitrary destinations. QUIC servers SHOULD NOT be
+deployed in networks that also have inadequately secured UDP
+endpoints.  Although it is not generally possible to ensure that
+clients are not co-located with vulnerable endpoints, this version of
+QUIC does not allow servers to migrate, thus preventing spoofed
+migration attacks on clients.  Any future extension which allows
+server migration MUST also define countermeasures for forgery attacks.
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:6455]:
```suggestion
endpoints.

Although it is not generally possible to ensure that
clients are not co-located with vulnerable endpoints, this version of
QUIC does not allow servers to migrate, thus preventing spoofed
migration attacks on clients.  Any future extension which allows
server migration MUST also define countermeasures for forgery attacks.
```

**Review by martinthomson — APPROVED** [review]
Thanks for writing this up.  My only comment is that this is two points and therefore two paragraphs.

**Comment by martinthomson** [comment]
This is already in the security considerations section, so we're good.  Though it's hard to see, because the section is already so large.

**Review by janaiyengar — APPROVED** [review]

**Pull request merged**